### ISSUE TITLE
No drivers available in the "Create connection" wizzard

### ISSUE THREAD

**Issue by acarstoiu** [dbeaver/dbeaver#23155]
### Description

I freshly installed the DBeaver plug-in within a freshly installed Eclipse _4.31_. Actually the original intention was to update just the Eclipse platform, but ended up in a total clean-up in an attempt to make it work.

The current problem is that **no drivers are available** in the "Driver Manager" dialog window and the "Create connection" wizzard. I even tried using a clean new workspace (and a new project within it), to no result.

There's nothing in the `dbeaver-debug.log` file and in the workspace `.log` fIle I can see just this missing logging implementation error:

```
!ENTRY org.eclipse.ui 4 0 2024-03-18 11:21:52.912
!MESSAGE Unhandled event loop exception
!STACK 0
java.lang.NoClassDefFoundError: org/slf4j/spi/LocationAwareLogger
	at org.apache.commons.logging.impl.Slf4jLogFactory.lambda$getInstance$0(Slf4jLogFactory.java:288)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at org.apache.commons.logging.impl.Slf4jLogFactory.getInstance(Slf4jLogFactory.java:286)
	at org.apache.commons.logging.impl.Slf4jLogFactory.getInstance(Slf4jLogFactory.java:281)
	at org.apache.commons.logging.LogFactory.getLog(LogFactory.java:987)
	at org.apache.commons.jexl3.internal.Engine.<init>(Engine.java:159)
	at org.apache.commons.jexl3.JexlBuilder.create(JexlBuilder.java:388)
	at org.jkiss.dbeaver.model.impl.AbstractDescriptor.parseExpression(AbstractDescriptor.java:134)
	at org.jkiss.dbeaver.model.navigator.meta.DBXTreeNode.<init>(DBXTreeNode.java:77)
	at org.jkiss.dbeaver.model.navigator.meta.DBXTreeFolder.<init>(DBXTreeFolder.java:79)
	at org.jkiss.dbeaver.registry.DataSourceProviderDescriptor.loadTreeNode(DataSourceProviderDescriptor.java:535)
	at org.jkiss.dbeaver.registry.DataSourceProviderDescriptor.loadTreeChildren(DataSourceProviderDescriptor.java:516)
	at org.jkiss.dbeaver.registry.DataSourceProviderDescriptor.loadTreeNode(DataSourceProviderDescriptor.java:595)
	at org.jkiss.dbeaver.registry.DataSourceProviderDescriptor.loadTreeChildren(DataSourceProviderDescriptor.java:516)
	at org.jkiss.dbeaver.registry.DataSourceProviderDescriptor.loadTreeNode(DataSourceProviderDescriptor.java:595)
	at org.jkiss.dbeaver.registry.DataSourceProviderDescriptor.loadTreeChildren(DataSourceProviderDescriptor.java:516)
	at org.jkiss.dbeaver.registry.DataSourceProviderDescriptor.loadTreeNode(DataSourceProviderDescriptor.java:595)
	at org.jkiss.dbeaver.registry.DataSourceProviderDescriptor.loadTreeChildren(DataSourceProviderDescriptor.java:516)
	at org.jkiss.dbeaver.registry.DataSourceProviderDescriptor.loadTreeNode(DataSourceProviderDescriptor.java:595)
	at org.jkiss.dbeaver.registry.DataSourceProviderDescriptor.loadTreeChildren(DataSourceProviderDescriptor.java:516)
	at org.jkiss.dbeaver.registry.DataSourceProviderDescriptor.loadTreeInfo(DataSourceProviderDescriptor.java:447)
	at org.jkiss.dbeaver.registry.DataSourceProviderDescriptor.<init>(DataSourceProviderDescriptor.java:109)
	at org.jkiss.dbeaver.registry.DataSourceProviderRegistry.loadExtensions(DataSourceProviderRegistry.java:123)
	at org.jkiss.dbeaver.registry.DataSourceProviderRegistry.getInstance(DataSourceProviderRegistry.java:64)
	at org.jkiss.dbeaver.registry.BasePlatformImpl.getDataSourceProviderRegistry(BasePlatformImpl.java:320)
	at org.jkiss.dbeaver.ui.controls.bool.BooleanStyleSet.installStyleChangeListener(BooleanStyleSet.java:156)
	at org.jkiss.dbeaver.ui.controls.ObjectViewerRenderer.<init>(ObjectViewerRenderer.java:114)
	at org.jkiss.dbeaver.ui.controls.ViewerColumnController$1.<init>(ViewerColumnController.java:118)
	at org.jkiss.dbeaver.ui.controls.ViewerColumnController.<init>(ViewerColumnController.java:118)
	at org.jkiss.dbeaver.ui.navigator.project.ProjectExplorerView.createColumns(ProjectExplorerView.java:124)
	at org.jkiss.dbeaver.ui.navigator.project.ProjectExplorerView.lambda$0(ProjectExplorerView.java:95)
	at org.jkiss.dbeaver.ui.UIExecutionQueue.executeInUI(UIExecutionQueue.java:68)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:5040)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4520)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1151)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1042)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:152)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:639)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:546)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:173)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:152)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:651)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:588)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1459)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1432)
Caused by: java.lang.ClassNotFoundException: org.slf4j.spi.LocationAwareLogger cannot be found by org.apache.commons.commons-logging_1.3.0
	at org.eclipse.osgi.internal.loader.BundleLoader.generateException(BundleLoader.java:562)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:557)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:434)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:174)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 58 more
```
If that is the **actual cause**, what's wrong with SLF4J?

### DBeaver Version

Community Edition DBeaver IDE 24.0.0.202403031740

### Operating System

Linux

**Comment by acarstoiu** [comment]
Also why can't I type _f i l e_ (without spaces) in a comment?! It is automatically replaced with _file_ 😲

**Comment by serge-rider** [comment]
Did you install any other Eclipse extensions but DBeaver?

org.apache.commons.commons-logging_1.3.0 is not a part of Eclipse or DBeaver. And it has broken dependencies.

Btw Eclipse 2024-03 seems to be a buggy version. We still cannot migrate to it because of number of issues on MacOS and Linux.

**Comment by acarstoiu** [comment]
Well, it looks like you're using the `org.apache.commons.jexl3.JexlBuilder` class which in turn needs to do some logging, thus triggering the error. Maybe _its_ dependencies are wrong.
And yes, I obtained this after uninstalling **all** other plugins and restarting - I was running DBeaver CE directly over the Eclipse platform.

I downgraded the Eclipse platform to _4.30_ in order to get a working environment. Eclipse platform _4.31_ is way fatter than _4.30_ (140 MB vs 90 MB), so it already seems fishy.
